### ISSUE TITLE
PVC creation failure for job caches

### ISSUE THREAD

**Issue by radu-matei** [brigadecore/brigade#917]
Trying to run a script with a job cache fails:

```
const { events, Job } = require("brigadier")

events.on("exec", (e) => {
    var job = new Job("cacher", "alpine:3.7")
    job.cache.enabled = true

    job.tasks = [
        "echo " + e.buildID + " >> /mnt/brigade/cache/jobs.txt",
        "cat /mnt/brigade/cache/jobs.txt"
    ]

    job.run()
})

```

```
     message: 'PersistentVolumeClaim "-cacher" is invalid: metadata.name: Invalid value: "-cacher": a DNS-1123 subdomain must consist of lower case alphanumeric characters, \'-\' or \'.\', and must start and end with an alphanumeric character (e.g. \'example.com\', regex used for validation is \'[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*\')',

```

**Comment by radu-matei** [comment]
Related - function to get the cache PVC name (from [here](https://github.com/brigadecore/brigade/blob/master/brigade-worker/src/k8s.ts#L467)):

```
  /**
   * cacheName returns the name of this job's cache PVC.
   */
  protected cacheName(): string {
    // The Kubernetes rules on pvc names are stupid^b^b^b^b strict. Name must
    // be DNS-like, and less than 64 chars. This rules out using project ID,
    // project name, etc. For now, we use project name with slashes replaced,
    // appended to job name.
    return `${this.project.name.replace(/[.\/]/g, "-")}-${
      this.job.name
      }`.toLowerCase();
  }
```

**Comment by dgkanatsios** [comment]
I've isolated the error, seems to be caused by [this line](https://github.com/brigadecore/brigade/blob/502eb908975c4cc3d516708c7a0adcb422845c75/brigade-worker/src/k8s.ts#L1009). Is there any reason that project name is set to the repo name? I'd be tempted to replace it with `secret.metadata.annotations["projectName"]`. Any objections?

**Comment by dgkanatsios** [comment]
Fixed by #892
